source/kmwbase.js

```diff
@@ -213,7 +213,8 @@ var __BUILD__ = 300;
       var func=util.events[event][i],result=false;
       try { result=func(params); }
       catch(strExcept) { result=false; }//don't know whether to use true or false here      
-      if(!result) { util.currentEvents.pop(); return false; }
+      if(result === false) {return false;}
+	  else {util.currentEvents.pop();}
```

Comment:
I think you are correct there. Looking back at my change, I think I only added the else statement because I misread how the function worked.

Unfortunately you won't be able to properly test behavior when multiple events are defined because that didn't work in the first place. Most UI functions called by events have no return value which meant that (!result) evaluated to true. This would make the callEvent function return before it finished executing all events in the array.

Using util.currentEvents.pop() didn't break anything in my own use of the keyboard, but I think that's only because the handlers are called directly from the events array instead of the currentEvents array. I'll change my local implementation to what you've suggested above and let you know if anything breaks. I suspect it will work fine though.